### PR TITLE
remove `.hasOwnProperty` check for shoebox keys

### PR DESCRIPTION
**Pull request by bekzod**
no need to check for `.hasOwnProperty` if shoebox has empty proto https://github.com/ember-fastboot/ember-cli-fastboot/pull/417

### CODE REVIEW

**Comment by stefanpenner** [comment]
these two pr #417 may need to be the same one?

**Comment by bekzod** [comment]
is possible to make same PR into 2 repos ?

**Comment by kellyselden** [comment]
This is going to be a tricky change, because once one repo gets the change, older versions of the other repo will not work with it. I suggest first checking if hasOwnProperty exists, then use it if it is. Then we can make a similar change in the other repo, release new RCs, then go about removing these blocks.

**Comment by kratiahuja** [comment]
We also have a manifest version to avoid such issues ;)

**Comment by kellyselden** [comment]
Ah I forgot about that!

**Comment by kratiahuja** [comment]
I prefer we punt this change to future. The change is small that it does not warrant for a schema version bump. In future when we do a schema version bump, we can do this change implicitly. thoughts?

**Comment by stefanpenner** [comment]
> thoughts?

Should we keep an open issue somewhere for pending things we want to do for a schema change? That way when we do do one, we have a backlog of changes?

File: src/ember-app.js [hunk 1]
```diff
@@ -435,8 +435,6 @@ function createShoebox(doc, fastbootInfo) {
   if (!shoebox) { return; }
 
   for (let key in shoebox) {
-    if (!shoebox.hasOwnProperty(key)) { continue; }
```
Comment (stefanpenner) [inline src/ember-app.js]: to future proof this. We could use `Object.hasOwnProperty(showbox, key)` instead?

Comment (bekzod) [inline src/ember-app.js]: sounds awesome, so it can be merged now :)

Comment (bekzod) [inline src/ember-app.js]: using `Object.hasOwnProperty(showbox, key)` now, I think now both PR's can be merged

**Comment by kratiahuja** [comment]
Yes I would love for an issue to be opened and we can label it to pick this change implicitly.

**Comment by kellyselden** [comment]
@kratiahuja With the recent changes, this has no consequence with merging. It's the other PR we have to worry about https://github.com/ember-fastboot/ember-cli-fastboot/issues/417.

**Comment by kratiahuja** [comment]
Ok I am good with it then. Didn't see the changes before.

**Pull request merged**